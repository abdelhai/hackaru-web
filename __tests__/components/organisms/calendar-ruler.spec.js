import Factory from '@/__tests__/__setups__/factory';
import CalendarRuler from '@/components/organisms/calendar-ruler';

describe('CalendarRuler', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(CalendarRuler, {
      provide: {
        pxPerMin: 40 / 60
      },
      propsData: {
        color: '#ff0'
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when has top', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setProps({ top: 50 });
    });

    it('render correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when showTime is true', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setProps({ top: 50, showTime: true });
    });

    it('render correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
