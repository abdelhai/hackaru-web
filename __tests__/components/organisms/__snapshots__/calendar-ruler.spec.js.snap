// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarRuler render correctly 1`] = `<!---->`;

exports[`CalendarRuler when has top render correctly 1`] = `
<section
  class="ruler"
  name="fade"
  style="margin-top: 50px;"
>
  <span
    class="line"
    style="background: rgb(255, 255, 0);"
  />
   
  <!---->
</section>
`;

exports[`CalendarRuler when showTime is true render correctly 1`] = `
<section
  class="ruler"
  name="fade"
  style="margin-top: 50px;"
>
  <span
    class="line"
    style="background: rgb(255, 255, 0);"
  />
   
  <time
    style="background: rgb(255, 255, 0);"
  >
    
      01:15
    
  </time>
</section>
`;
